test/tz_fake.py

```diff
@@ -0,0 +1,37 @@
+import pytz
+from contextlib import contextmanager
+import time
+import os
+import pytest
+import tzlocal
+
+
+@contextmanager
+def fake_timezone_ctx(tz):
+    """
+    Switch to a locally-known timezone specified by `tz`.
+    On exit, restore the previous timezone.
+    If `tz` is `None`, do nothing.
+    """
+    if tz is None:
+        yield
+        return
+
+    if not hasattr(time, "tzset"):
```

Comment:
We should document in a comment when this is the case (ie `non-Unix`)

Reply:
Done.